### PR TITLE
feat: add `TorV3` support

### PR DESCRIPTION
**Pull request by oleonardolima**

### CODE REVIEW

**Comment by rustaceanrob** [comment]
Ran this locally and LGTM. Is this ready to go?

**Comment by oleonardolima** [comment]
> Ran this locally and LGTM. Is this ready to go?

Did you manage to connect to an onion address peer ?

I didn't get any yet, might've broken something locally.

**Comment by rustaceanrob** [comment]
The example uses a preset peer and only connects to one peer. I'll change it locally to try to find one at random 😂

File: src/peers/tor.rs [hunk 1]
```diff
@@ -38,31 +39,43 @@ impl NetworkConnector for TorConnection {
...
+                        .map_err(|_e| PeerError::ConnectionFailed)?
                 }
-                AddrV2::TorV2(_onion) => Err(PeerError::UnreachableSocketAddr),
-                AddrV2::TorV3(_onion) => Err(PeerError::UnreachableSocketAddr),
-                _ => Err(PeerError::UnreachableSocketAddr),
-            }
+                AddrV2::TorV3(pk) => {
+                    let hs_id = HsId::from(pk);
+                    let tor_addr = TorAddr::from(hs_id.to_string())
+                        .map_err(|_e| PeerError::UnreachableSocketAddr)?;
```
Comment (rustaceanrob) [inline src/peers/tor.rs:64]: Needs to be:

```rust
TorAddr::from(format!("{}:{}", hs_id.to_string(), port))
```

**Review by rustaceanrob — COMMENTED** [review]
Got the bug. Was able to connect to onion addrs!

File: src/peers/tor.rs (hunk 1 above)
Comment (oleonardolima) [reply src/peers/tor.rs:61]: Thanks, I amended the fix to 2ddd0f2

**Comment by rustaceanrob** [comment]
Works. OK if I merge?

**Pull request merged**